org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/JDTUtilsTest.java

```diff
@@ -229,4 +230,22 @@ public void testIsFolder() throws Exception {
 		assertNotNull(JDTUtils.findFile(uriFile.toString()));
 		assertNotNull(JDTUtils.findFolder(uriFolder.toString()));
 	}
+
+	@Test
+	public void testGetFileOrFolder() throws Exception {
+		// For: https://github.com/eclipse/eclipse.jdt.ls/issues/1137
+		IProject project = WorkspaceHelper.getProject(ProjectsManager.DEFAULT_PROJECT_NAME);
+
+		File parentFolder = new File(project.getLocation().toString(), "/src/org/eclipse");
+		IResource parent = JDTUtils.getFileOrFolder(parentFolder.toURI().toString());
+		assertTrue(parent instanceof IFolder);
+		final int originalMemberNum = ((IFolder) parent).members().length;
+
+		File newPackage = new File(project.getLocation().toString(), "/src/org/eclipse/testGetFileOrFolder");
+		newPackage.mkdirs();
```

Comment:
will the new package be cleaned up after the test done?

Reply:
Yes, the super class will do this after tests:

https://github.com/eclipse/eclipse.jdt.ls/blob/fc6b40e58808d23a324628a1e79e50f117c534de/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/AbstractWorkspaceTest.java#L27-L30